### ISSUE TITLE
Problems with receiving OrderBook updates

### ISSUE THREAD

**Issue by FlashPlayer13** [JKorf/Bitfinex.Net#92]
Last few days there are problems with receiving book updates. I am subscribing like this:

```
var  _socketClient = new BitfinexSocketClient(new BitfinexSocketClientOptions {ApiCredentials = new ApiCredentials(key, secret), ReconnectInterval = new TimeSpan(0, 1, 0), SocketNoDataTimeout = new TimeSpan(0, 1, 0)});
var tickerLocal = "tXRPUSD";
_socketClient.SubscribeToBookUpdates(tickerLocal, Precision.PrecisionLevel0, Frequency.Realtime, 1, data => OrderBookUpdatesHandler(tickerLocal, data));
```

but no data drops into handler. Something changed on bitfinex or?

**Comment by JKorf** [comment]
Hi, if I test your code locally the problem seems to be the length parameter which is set to 1. It should be either 25 or 100. I don't think 1 has ever worked? 
To make sure the subscription is successful and what the error might be you should check the result of the subscribe call: 
````C#
            var subResult = _socketClient.SubscribeToBookUpdates(tickerLocal, Precision.PrecisionLevel0, Frequency.Realtime, 25, data => OrderBookUpdatesHandler(tickerLocal, data));
            if (!subResult.Success)
            {
                Console.WriteLine($"Failed to subscribe: {subResult.Error}");
                return;
            }
````
![image](https://user-images.githubusercontent.com/8448172/62461688-c11dcd80-b785-11e9-9a9c-b5702b5427b9.png)

Could you try with length = 25?

**Comment by FlashPlayer13** [comment]
Thank you for your help.
I tried 25 and now it works. 
Actually i just updated few days ago from 2.0.10 version to latest one (2.1.3) and looks like some things work different now. For example, this subscribe worked with limit=1 on version 2.0.10 so i always had best bid/ask here.
Also i found one more difference (maybe this is a problem, maybe not).
On previous version when order was placed i received a message into this callback (OrderHandler):
`_socketClient.SubscribeToTradingUpdates(OrderHandler, TradeHandler, PositionHandler)`
But now looks like when order is placed - no message drops here. Is this a correct behavior now?

**Comment by JKorf** [comment]
Huh that's intresting, the documentation actually says the length should be either 25 or 100: https://docs.bitfinex.com/v2/reference#ws-public-order-books, but I guess that's not enforced.

In regards of the trade update, it still works for me with the latest version. If you have issues you can turn on debug logging (LogVerbosity = LogVerbosity.Debug in client options) and check what's happening.

**Comment by FlashPlayer13** [comment]
I will try to describe my problem clearly:
When i'm placing a new order i expect to get an update with this subscription:
`_socketClient.SubscribeToTradingUpdates`
like BitfinexSocketEvent with ACTIVE new order. But for some reason i'm not receving any message about new order here anymore. Earlier (2.0.10) it worked well. Earlier this callback got all messages about life cycle of order: from placing till executing/cancelling. But now placing message doesn't show up.
I tried with LogVerbosity = LogVerbosity.Debug and here are logs that i got (after i placed one order):

> 2019.08.05 17:00:12:049 | Debug | Socket 1 received data: [85975,"te",[383266085,1565013611430,-25.94,0.32675]]
> 2019.08.05 17:00:12:064 | Debug | Socket 1 received data: [149743,[0.32807,0,-1]]
> 2019.08.05 17:00:12:066 | Debug | Socket 1 received data: [149743,[0.32675,2,4663.7350042]]
> 2019.08.05 17:00:12:068 | Debug | Socket 1 received data: [149743,[0.32708,1,-2337.6443783]]
> 2019.08.05 17:00:12:069 | Debug | Socket 1 received data: [149743,[0.32709,1,-16314.11715164]]
> 2019.08.05 17:00:12:072 | Info | Going to place order
> 2019.08.05 17:00:12:080 | Debug | Socket 1 received data: [85975,"tu",[383266085,1565013611430,-25.94,0.32675]]
> 2019.08.05 17:00:12:091 | Debug | Socket 1 sending data: [0,"on",null,{"type":"LIMIT","symbol":"tXRPUSD","amount":"-12.18100981","price":"0.32838","cid":1}]
> 2019.08.05 17:00:12:167 | Debug | Socket 1 received data: [0,"n",[1565013611.561,"on-req",null,null,[29435771598,null,1,"tXRPUSD",1565013611560,1565013611560,-12.18100981,-12.18100981,"LIMIT",null,null,null,0,"ACTIVE",null,null,0.32838,0,0,0,null,null,null,0,null,null,null,null,"API>BFX",null,null,null],null,"SUCCESS","Submitting limit sell order for -12.18100981 XRP."]]
> 2019.08.05 17:00:12:168 | Debug | Message not handled: [
>   0,
>   "n",
>   [
>     1565013611.561,
>     "on-req",
>     null,
>     null,
>     [
>       29435771598,
>       null,
>       1,
>       "tXRPUSD",
>       1565013611560,
>       1565013611560,
>       -12.18100981,
>       -12.18100981,
>       "LIMIT",
>       null,
>       null,
>       null,
>       0,
>       "ACTIVE",
>       null,
>       null,
>       0.32838,
>       0,
>       0,
>       0,
>       null,
>       null,
>       null,
>       0,
>       null,
>       null,
>       null,
>       null,
>       "API>BFX",
>       null,
>       null,
>       null
>     ],
>     null,
>     "SUCCESS",
>     "Submitting limit sell order for -12.18100981 XRP."
>   ]
> ]
> 2019.08.05 17:00:12:174 | Debug | Socket 1 received data: [149743,[0.32649,0,1]]
> 2019.08.05 17:00:12:175 | Debug | Socket 1 received data: [149743,[0.32604,0,1]]
> 2019.08.05 17:00:12:176 | Debug | Socket 1 received data: [149743,[0.32709,0,-1]]
> 2019.08.05 17:00:12:177 | Debug | Socket 1 received data: [149743,[0.32723,0,-1]]
> 2019.08.05 17:00:12:178 | Debug | Socket 1 received data: [149743,[0.32742,0,-1]]
> 2019.08.05 17:00:12:179 | Debug | Socket 1 received data: [149743,[0.32758,0,-1]]
> 2019.08.05 17:00:12:180 | Debug | Socket 1 received data: [149743,[0.32761,0,-1]]
> 2019.08.05 17:00:12:181 | Debug | Socket 1 received data: [149743,[0.32667,1,4482.74136799]]
> 2019.08.05 17:00:12:182 | Debug | Socket 1 received data: [149743,[0.32617,1,9194.83863]]
> 2019.08.05 17:00:12:183 | Debug | Socket 1 received data: [149743,[0.32613,1,12235.59640885]]
> 2019.08.05 17:00:12:184 | Debug | Socket 1 received data: [149743,[0.32707,1,-16314.11715164]]
> 2019.08.05 17:00:12:185 | Debug | Socket 1 received data: [149743,[0.32737,1,-4415.80907196]]
> 2019.08.05 17:00:12:186 | Debug | Socket 1 received data: [149743,[0.32744,1,-12235.59640886]]
> 2019.08.05 17:00:12:186 | Debug | Socket 1 received data: [149743,[0.32807,1,-14000]]
> 2019.08.05 17:00:12:187 | Debug | Socket 1 received data: [149743,[0.32809,1,-16822]]
> 2019.08.05 17:00:12:188 | Debug | Socket 1 received data: [0,"on",[29435771598,null,1,"tXRPUSD",1565013611560,1565013611563,-12.18100981,-12.18100981,"LIMIT",null,null,null,0,"ACTIVE",null,null,0.32838,0,0,0,null,null,null,0,0,null,null,null,"API>BFX",null,null,null]]
> 

I see "Message not handled" message, but not sure what it means. Hope you can help to understand what is happening.

**Comment by JKorf** [comment]
Ah, I've found the problem. I initially tested your issue by placing an order via the rest api, which results in the updates you'd expect. However, when placing them via the socket the updates were missing. I've fixed it in the latest version 2.1.5 which should be available in a few minutes.

**Comment by FlashPlayer13** [comment]
Thank you! 
Now it works like before - problem solved!)
